ssl/src/main/java/io/crate/protocols/ssl/SslContextProviderImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.ssl;
+
+import io.netty.handler.ssl.SslContext;
+import org.elasticsearch.common.settings.Settings;
+
+import java.lang.reflect.InvocationTargetException;
+
+class SslContextProviderImpl implements SslContextProvider {
```

Comment:
This should go into the ssl-impl enterprise module and directly access `Sslconfiguration` - without dynamic class loading.